src/bsoncxx/json.hpp

```diff
@@ -34,18 +34,39 @@ BSONCXX_INLINE_NAMESPACE_BEGIN
 extern const bool k_silence_doxygen;
 
 ///
-/// Converts a BSON document to a JSON string.
+/// An enumeration of the types of Extended JSON that the to_extended_json function accepts
+///
+/// The values are:
+///   k_canonical: to produce Canonical Extended JSON
+///   k_relaxed: to produce Relaxed Extended JSON
+enum class ExtendedJsonMode { k_canonical, k_relaxed };
+
+///
+/// Converts a BSON document to a JSON string, in legacy extended format.
 ///
 /// @param view
 ///   A valid BSON document.
 ///
 /// @throws bsoncxx::exception with error details if the conversion failed.
 ///
-/// @returns A JSON string.
+/// @returns A legacy extended JSON string.
 ///
 BSONCXX_API std::string BSONCXX_CALL to_json(document::view view);
 
 ///
+/// Converts a BSON document to a JSON string, in extended format.
+///
+/// @param view
+///   A valid BSON document.
+///
+/// @throws bsoncxx::exception with error details if the conversion failed.
+///
+/// @returns A extended JSON string.
+///
+BSONCXX_API std::string BSONCXX_CALL
+to_extended_json(document::view view, ExtendedJsonMode mode = ExtendedJsonMode::k_relaxed);
```

Comment:
I wouldn't add a default. They bind statically in the caller. It made sense to add a default when we were considering adding the enum to `to_json` as an API non-break, but as this is a new function I don't see an upside.

Users can easily write their own wrapper function to default in a value if they choose.

Reply:
Sorry, nm. I misread that. I'll remove the default.